CRM/Civisocial/BAO/CivisocialUser.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/*
++--------------------------------------------------------------------+
+| CiviCRM version 4.6                                                |
++--------------------------------------------------------------------+
+| Copyright CiviCRM LLC (c) 2004-2015                                |
++--------------------------------------------------------------------+
+| This file is a part of CiviCRM.                                    |
+|                                                                    |
+| CiviCRM is free software; you can copy, modify, and distribute it  |
+| under the terms of the GNU Affero General Public License           |
+| Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+|                                                                    |
+| CiviCRM is distributed in the hope that it will be useful, but     |
+| WITHOUT ANY WARRANTY; without even the implied warranty of         |
+| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+| See the GNU Affero General Public License for more details.        |
+|                                                                    |
+| You should have received a copy of the GNU Affero General Public   |
+| License and the CiviCRM Licensing Exception along                  |
+| with this program; if not, contact CiviCRM LLC                     |
+| at info[AT]civicrm[DOT]org. If you have questions about the        |
+| GNU Affero General Public License or the licensing of CiviCRM,     |
+| see the CiviCRM license FAQ at http://civicrm.org/licensing        |
++--------------------------------------------------------------------+
+ */
+/**
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2015
+ *
+ */
+class CRM_Civisocial_BAO_CivisocialUser {
+
+  public static function get($params) {
+    $result = array();
+    $instance = new CRM_Civisocial_DAO_CivisocialUser();
+    if (!empty($params)) {
+      $fields = $instance->fields();
+      foreach ($params as $key => $value) {
+        if (isset($fields[$key])) {
+          $instance->$key = $value;
+        }
+      }
+    }
+    $instance->find();
+    while ($instance->fetch()) {
+      $row = array();
+      $instance->storeValues($instance, $row);
+      $result[$row['id']] = $row;
+    }
+    return $result;
+  }
+
+  /**
+   * Create social user
+   *
+   * @param string $params
+   */
+  public static function create($params) {
+    $className = 'CRM_Civisocial_DAO_CivisocialUser';
```

Comment:
since this function will use single class i.e 'CRM_Civisocial_DAO_CivisocialUser' to insert values into db so it would be good to use it directly at line 64 as new CRM_Civisocial_DAO_CivisocialUser() rather storing it in a variable